include/ignition/rendering/Utils.hh

```diff
@@ -30,6 +35,36 @@ namespace ignition
     // Inline bracket to help doxygen filtering.
     inline namespace IGNITION_RENDERING_VERSION_NAMESPACE {
     //
+    /// \brief Retrieve the first point on a surface in the 3D scene hit by a
+    /// ray cast from the given 2D screen coordinates.
+    /// \param[in] _screenPos 2D coordinates on the screen, in pixels.
+    /// \param[in] _camera User camera
+    /// \param[in] _rayQuery Ray query for mouse clicks
+    /// \param[inout] _rayResult Ray query result
+    /// \param[inout] _maxDistance maximum distance to check the collision
+    /// \return 3D coordinates of a point in the 3D scene.
+    IGNITION_RENDERING_VISIBLE
+    math::Vector3d ScreenToScene(
+        const math::Vector2i &_screenPos,
+        const CameraPtr &_camera,
+        const RayQueryPtr &_rayQuery,
+        RayQueryResult &_rayResult,
```

Comment:
ok thanks; I really like having the option of a `RayQueryResult&` in the API, but it may be a hassle for users who don't need it. I would also include an overload without it:

~~~
math::Vector3d ScreenToScene(
         const math::Vector2i &_screenPos,
         const CameraPtr &_camera,
         const RayQueryPtr &_rayQuery,
         float maxDistance = 10.0)
{
  RayQueryResult rayResult;
  return this->ScreenToScene(_screenPos, _camera, _rayQuery, rayResult, maxDistance);
}
~~~

Reply:
https://github.com/ignitionrobotics/ign-rendering/pull/363/commits/c3f358e4bc9697fe861c8857aa5d3c04d3a31c1d